core/src/main/java/org/infinispan/xsite/irac/IracResponseCollector.java

```diff
@@ -12,83 +13,117 @@
 import org.infinispan.util.logging.LogFactory;
 import org.infinispan.xsite.status.DefaultTakeOfflineManager;
 
+import net.jcip.annotations.GuardedBy;
+
 /**
  * A response collector for a single key update.
  * <p>
  * This class extends {@link CompletableFuture}. It is completed when all sites involved in the updated have replied (or
```

Comment:
The javadoc is lying: it extends `Runnable`

Reply:
I think it is also not about single keys anymore

Reply:
I've updated the javadoc.